### PR TITLE
feat: add xRoute and xPolicy support for egctl x status

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

- add support for `xRoute` and `xPolicy` input types
- display `Parents` and `Ancestors` for `xRoute` or `xPolicy`
- get rid of all the reflect errors handling, since we can promise all these fields exist

some usage example:

image

- the kind name in ParentRef has been changed to lower case

**Which issue(s) this PR fixes**:

follow-up of #2631

### CODE REVIEW

File: internal/cmd/egctl/status.go [hunk 1]
```diff
@@ -22,13 +22,28 @@ import (
...
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
 )
 
-// supportedTypes list all the resource types that status command supports.
-var supportedTypes = []string{
-	"GatewayClass", "Gateway", "HTTPRoute", "GRPCRoute",
-	"TLSRoute", "TCPRoute", "UDPRoute", "BackendTLSPolicy",
-	"BackendTrafficPolicy", "ClientTrafficPolicy", "EnvoyPatchPolicy", "SecurityPolicy",
+var (
+	supportedXRouteTypes = []string{
```
Comment (arkodg) [inline internal/cmd/egctl/status.go:29]: lets bring this up in the community meeting to get consensus on whether to introduce `xRoute` and `xPolicy` here or not

Comment (shawnh2) [reply internal/cmd/egctl/status.go:29]: The idea of implementing `xRoute` and `xPolicy` for `egctl x status` is actually from #2028 (the `Managed Resources Status` section).

Comment (arkodg) [reply internal/cmd/egctl/status.go:29]: resolving this one, should be fine to add this in

**Review by arkodg — DISMISSED** [review]
LGTM thanks !

**Review by zhaohuabing — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**